Common/Data/Auxiliary/MapFileRow.cs

```diff
@@ -70,7 +87,14 @@ public static IEnumerable<MapFileRow> Read(string path)
         public static MapFileRow Parse(string line)
         {
             var csv = line.Split(',');
-            return new MapFileRow(DateTime.ParseExact(csv[0], DateFormat.EightCharacter, null), csv[1]);
+            var primaryExchange = PrimaryExchange.UNKNOWN;
+            if (csv.Length == 3)
+            {
+                primaryExchange = Exchanges.GetPrimaryExchange(Convert.ToChar(csv[2], CultureInfo.InvariantCulture));
+            }
+
+            return new MapFileRow(DateTime.ParseExact(csv[0], DateFormat.EightCharacter, null), csv[1],
+                Convert.ToChar((byte) primaryExchange));
```

Comment:
Could we skip the new constructor if we just passed in `primaryExchange` directly?